src/cli/CMakeLists.txt

```diff
@@ -59,6 +59,45 @@ install(TARGETS keepassxc-cli
         BUNDLE DESTINATION . COMPONENT Runtime
         RUNTIME DESTINATION ${CLI_INSTALL_DIR} COMPONENT Runtime)
 
+    if(APPLE AND WITH_APP_BUNDLE)
+        set(CLI_BINARY_DIR "${CMAKE_BINARY_DIR}/src/cli/keepassxc-cli")
+        set(CLI_APP_DIR "KeePassXC.app/Contents/MacOS/keepassxc-cli")
+        add_custom_command(TARGET keepassxc-cli
+                           POST_BUILD
+                           COMMAND ${CMAKE_COMMAND} -E copy ${CLI_BINARY_DIR} ${CLI_APP_DIR}
+                           WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src
+                           COMMENT "Copying keepassxc-cli inside the application")
+
+        add_custom_command(TARGET keepassxc-cli
+                           POST_BUILD
+                           COMMAND ${CMAKE_INSTALL_NAME_TOOL}
+                                -change /usr/local/opt/qt/lib/QtCore.framework/Versions/5/QtCore
```

Comment:
You cannot use static paths for Qt libraries. User might have installed them manually or through MacPorts etc.. You can use Qt5_PREFIX instead:
```
-change ${Qt5_PREFIX}/lib/QtCore.framework/Versions/5/QtCore
                 "@executable_path/../Frameworks/QtCore.framework/Versions/5/QtCore"
```

Reply:
Where exactly is ${Qt5_PREFIX} defined? When I tried replacing `/usr/local/opt/qt` with `${Qt5_PREFIX}` for each of the linking changes needed, built and checked the linking of the binary again, the linking was not changed from the static paths.

Or is ${Qt5_PREFIX} manually defined by the user?

Reply:
Also, which variable (if any) should I use to refer to the libsodium library?

Reply:
```
keepassxc/CMakeLists.txt:
310: get_filename_component(Qt5_PREFIX ${Qt5_DIR}/../../.. REALPATH)
```

Of course the path remains the same in your computer. For libsodium you could try using ${sodium_LIBRARY_DEBUG} and ${sodium_LIBRARY_RELEASE}.

Reply:
Thanks

I'm wondering the best way to go about this because currently the way I have the paths set in  `src/cli/CMakeLists.txt` is like this.

```
-change ${sodium_LIBRARY_RELEASE}
     "@executable_path/../Frameworks/libsodium.23.dylib"
-change /usr/local/opt/libsodium/lib/libsodium.23.dylib
    "@executable_path/../Frameworks/libsodium.23.dylib"
```

According to `build/CMakeLists.txt` `sodium_LIBRARY_RELEASE:FILEPATH=/usr/local/Cellar/libsodium/1.0.16/lib/libsodium.dylib` doesn't match up with which was linked by default for libsodium in `keepassxc-cli` on my system, which is `/usr/local/opt/libsodium/lib/libsodium.23.dylib (compatibility version 25.0.0, current version 25.0.0)`

Reply:
I think this is the best way yes. Also with `${Qt5_PREFIX}`. Looking the the proxy's CMakeList.txt, both prefix and fixed location is used to ensure it would work on different environments. `libyubikey` and `libykpers` should be also added to the list. At least for me `otool` shows them in the list also (using `WITH_XC_ALL=ON`).

For me this works OK:
<details>
<summary>Code block of app bundle</summary>

    if(APPLE AND WITH_APP_BUNDLE)
        set(CLI_BINARY_DIR "${CMAKE_BINARY_DIR}/src/cli/keepassxc-cli")
        set(CLI_APP_DIR "KeePassXC.app/Contents/MacOS/keepassxc-cli")
        list(GET YUBIKEY_LIBRARIES 0 YUBIKEY_CORE_LIBRARY)
        list(GET YUBIKEY_LIBRARIES 1 YUBIKEY_PERS_LIBRARY)
        add_custom_command(TARGET keepassxc-cli
                           POST_BUILD
                           COMMAND ${CMAKE_COMMAND} -E copy ${CLI_BINARY_DIR} ${CLI_APP_DIR}
                           WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src
                           COMMENT "Copying keepassxc-cli inside the application")

        add_custom_command(TARGET keepassxc-cli
                           POST_BUILD
                           COMMAND ${CMAKE_INSTALL_NAME_TOOL}
                                -change ${Qt5_PREFIX}lib/QtCore.framework/Versions/5/QtCore
                                    "@executable_path/../Frameworks/QtCore.framework/Versions/5/QtCore"
                                -change /usr/local/opt/qt/lib/QtCore.framework/Versions/5/QtCore
                                    "@executable_path/../Frameworks/QtCore.framework/Versions/5/QtCore"
                                -change ${Qt5_PREFIX}/lib/QtGui.framework/Versions/5/QtGui
                                    "@executable_path/../Frameworks/QtGui.framework/Versions/5/QtGui"
                                -change /usr/local/opt/qt/lib/QtGui.framework/Versions/5/QtGui
                                    "@executable_path/../Frameworks/QtGui.framework/Versions/5/QtGui"
                                -change ${Qt5_PREFIX}/lib/QtMacExtras.framework/Versions/5/QtMacExtras
                                    "@executable_path/../Frameworks/QtMacExtras.framework/Versions/5/QtMacExtras"
                                -change /usr/local/opt/qt/lib/QtMacExtras.framework/Versions/5/QtMacExtras
                                    "@executable_path/../Frameworks/QtMacExtras.framework/Versions/5/QtMacExtras"
                                -change ${Qt5_PREFIX}/lib/QtConcurrent.framework/Versions/5/QtConcurrent
                                    "@executable_path/../Frameworks/QtConcurrent.framework/Versions/5/QtConcurrent"
                                 -change /usr/local/opt/qt/lib/QtConcurrent.framework/Versions/5/QtConcurrent
                                    "@executable_path/../Frameworks/QtConcurrent.framework/Versions/5/QtConcurrent"
                                -change ${Qt5_PREFIX}/lib/QtCore.framework/Versions/5/QtCore
                                    "@executable_path/../Frameworks/QtCore.framework/Versions/5/QtCore"
                                -change /usr/local/opt/qt/lib/QtCore.framework/Versions/5/QtCore
                                    "@executable_path/../Frameworks/QtCore.framework/Versions/5/QtCore"
                                -change ${Qt5_PREFIX}/lib/QtNetwork.framework/Versions/5/QtNetwork
                                    "@executable_path/../Frameworks/QtNetwork.framework/Versions/5/QtNetwork"
                                -change /usr/local/opt/qt/lib/QtNetwork.framework/Versions/5/QtNetwork
                                    "@executable_path/../Frameworks/QtNetwork.framework/Versions/5/QtNetwork"
                                -change ${Qt5_PREFIX}/lib/QtWidgets.framework/Versions/5/QtWidgets
                                    "@executable_path/../Frameworks/QtWidgets.framework/Versions/5/QtWidgets"
                                -change /usr/local/opt/qt/lib/QtWidgets.framework/Versions/5/QtWidgets
                                    "@executable_path/../Frameworks/QtWidgets.framework/Versions/5/QtWidgets"
                                -change ${GCRYPT_LIBRARIES}
                                    "@executable_path/../Frameworks/libgcrypt.20.dylib"
                                -change /usr/local/opt/libgcrypt/lib/libgcrypt.20.dylib
                                    "@executable_path/../Frameworks/libgcrypt.20.dylib"
                                -change ${ARGON2_LIBRARIES}
                                    "@executable_path/../Frameworks/libargon2.1.dylib"
                                -change /usr/local/opt/argon2/lib/libargon2.1.dylib
                                    "@executable_path/../Frameworks/libargon2.1.dylib"
                                -change ${GPGERROR_LIBRARIES}
                                    "@executable_path/../Frameworks/libgpg-error.0.dylib"
                                -change /usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib
                                    "@executable_path/../Frameworks/libgpg-error.0.dylib"
                                -change ${sodium_LIBRARY_RELEASE}
                                    "@executable_path/../Frameworks/libsodium.23.dylib"
                                 -change /usr/local/opt/libsodium/lib/libsodium.23.dylib
                                    "@executable_path/../Frameworks/libsodium.23.dylib"
                                -change ${YUBIKEY_CORE_LIBRARY}
                                    "@executable_path/../Frameworks/libyubikey.0.dylib"
                                -change /usr/local/opt/libyubikey/lib/libyubikey.0.dylib
                                    "@executable_path/../Frameworks/libyubikey.0.dylib"
                                -change ${YUBIKEY_PERS_LIBRARY}
                                    "@executable_path/../Frameworks/libykpers-1.1.dylib"
                                -change /usr/local/opt/ykpers/lib/libykpers-1.1.dylib
                                    "@executable_path/../Frameworks/libykpers-1.1.dylib"
                           ${CLI_APP_DIR}
                           WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src
                           COMMENT "Changing linking of keepassxc-cli")
    endif()

</details>

Reply:
On my system I get this error after updating that section of src/cli/CMakeLists.txt to match yours.

```
Copying keepassxc-cli inside the application
Changing linking of keepassxc-cli
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: more than one input file specified (@executable_path/../Frameworks/libsodium.23.dylib and /usr/local/opt/libsodium/lib/libsodium.23.dylib)
Usage: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool [-change old new] ... [-rpath old new] ... [-add_rpath new] ... [-delete_rpath old] ... [-id name] input
make[2]: *** [src/cli/keepassxc-cli] Error 1
make[2]: *** Deleting file `src/cli/keepassxc-cli'
make[1]: *** [src/cli/CMakeFiles/keepassxc-cli.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Is there an easy way to list the full command line (including arguments) of install_name_tool? called by cmake?

If I remove this line:

```
                            -change ${sodium_LIBRARY_RELEASE}
                                "@executable_path/../Frameworks/libsodium.23.dylib"
```
My otool -L output is equivalent to yours.

Reply:
That libsodium variable was giving me some headaches too. I'm not sure if that line above is even necessary. I haven't found another way to include the libs.